s4e-web/src/app/errors/errors.component.ts

```diff
@@ -0,0 +1,47 @@
+import { ActivatedRoute } from '@angular/router';
+import { Component } from '@angular/core';
+import { ERROR_CODE_NOT_FOUND, ERROR_CODE_BAD_GATEWAY, ERROR_CODE_INTERNAL_SERVER_ERROR } from './errors.model';
+
+@Component({
+  selector: 's4e-errors',
+  templateUrl: './errors.component.html',
+  styleUrls: ['./errors.component.scss']
+})
+export class ErrorsComponent {
+  public actualError;
+  public backLink?: string;
+
+  constructor(private _activatedRoute: ActivatedRoute) {
+    this._activatedRoute.paramMap.subscribe((params) => {
+      this.backLink = params.get('backLink');
+
+      const actualErrorCode = +params.get('errorCode');
+      this.actualError = ErrorsComponent._getErrorBy(actualErrorCode);
+    });
```

Comment:
All subscribes, etc should be done in `ngOnInit` (implemet interface `OnInit` and `OnDestroy`.
Additionally do not forget to unsubscribe from all subscriptions. The easiest way to do this in the context of the component is to have it implement `OnDestroy` and use `.pipe(untilDestoryed(this)).subscribe(...)`
Check other components which subscribe to things.

Reply:
Next thing to be added to refactoring stuff, I'll update refactoring list with it. There is also another type of solution. Http Observable can be transformed into Promise, so there will be any need of subscription (just solved by using then()).